src/browser/setupWorker/glossary.ts

```diff
@@ -197,11 +198,13 @@ export interface StartOptions extends SharedOptions {
   findWorker?: FindWorker
 }
 
-export type StartReturnType = Promise<ServiceWorkerRegistration | undefined>
+export type StartReturnType = Promise<ServiceWorkerRegistration>
+
 export type StartHandler = (
   options: RequiredDeep<StartOptions>,
   initialOptions: StartOptions,
-) => StartReturnType
+) => DeferredPromise<ServiceWorkerRegistration>
```

Comment:
The internal return type is a deferred promise so we can reject the pending activation promise at any time (e.g. when calling `stop()`).